pkg/registry/apis/provisioning/jobs/pullrequest/worker.go

```diff
@@ -104,12 +107,12 @@ func (c *PullRequestWorker) Process(ctx context.Context,
 		return nil
 	}
 
-	fileInfo, err := prRepo.Read(ctx, f.Path, ref)
+	fileInfo, err := prRepo.Read(ctx, f.Path, options.Ref)
 	if err != nil {
 		return fmt.Errorf("read file: %w", err)
 	}
 
-	_, err = parser.Parse(ctx, fileInfo)
+	obj, err := parser.Parse(ctx, fileInfo)
```

Comment:
this now uses the parsed object properties rather than passing them as parameters